mlir/include/Catalyst/Driver/CompilerDriver.h

```diff
@@ -0,0 +1,115 @@
+// Copyright 2023 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include "mlir/IR/MLIRContext.h"
+#include "mlir/Support/LogicalResult.h"
+#include "llvm/ADT/SmallVector.h"
+#include "llvm/Support/raw_ostream.h"
+#include <filesystem>
+#include <vector>
```

Comment:
Please organize imports from most general to most specific (separated by an empty line):

- stdlib imports at the top
- external libraries like mlir/llvm
- internal imports from other parts of the codebase

(Make sure to check all other source files in this PR as well.)